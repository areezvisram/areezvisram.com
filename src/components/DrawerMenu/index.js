import React from 'react';
import { IconButton, Drawer, makeStyles, Box, useTheme } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import TabItem from '../NavigationItem/TabItem';

const useStyles = makeStyles((theme) => ({
    drawer: {
        backgroundColor: theme.palette.primary.main,
        width: theme.spacing(30),
        overflowX: 'hidden'
    },

    closeIcon: {
        color: theme.palette.primary.contrastText,
    },
}));

const DrawerMenu = ({ isDrawerOpen, handleDrawerOpen, tabInfo, tabComponent }) => {
    const classes = useStyles();
    const theme = useTheme();
    return (
        <Drawer
        variant="temporary"
        anchor={'right'}
        open={isDrawerOpen}
        onClose={handleDrawerOpen}
        ModalProps={{
            keepMounted: true,
        }}  
        classes={{ paper: classes.drawer }} 
        transitionDuration={450}
        >    
            <Box textAlign={'right'}> 
                <IconButton onClick={handleDrawerOpen}>
                    <CloseIcon className={classes.closeIcon}/>
                </IconButton>                
            </Box>
            <Box textAlign={'center'} marginTop={theme.spacing(0.5)}>
                {tabInfo.map(tab => <TabItem label={tab.label} route={tab.route} component={tabComponent} key={tab.value} />)}
            </Box>
        </Drawer> 
    )
}

export default DrawerMenu;