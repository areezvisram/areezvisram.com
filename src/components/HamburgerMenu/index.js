import React, { useState } from 'react';
import { IconButton, makeStyles } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import DrawerMenu from '../DrawerMenu';

const useStyles = makeStyles((theme) => ({
    iconButton: {
        color: theme.palette.primary.contrastText,        
    },
    menuIcon: {
        fontSize: theme.spacing(5)
    }
}));

const HamburgerMenu = () => {
    const classes = useStyles();
    const [isDrawerOpen, setIsDrawerOpen] = useState(false);

    const handleDrawerOpen = () => {             
        setIsDrawerOpen(!isDrawerOpen)
    }
    return (
        <div>
            <IconButton className={classes.iconButton} size={'small'} onClick={handleDrawerOpen}>
                <MenuIcon className={classes.menuIcon} />
            </IconButton>
                        
           <DrawerMenu isDrawerOpen={isDrawerOpen} handleDrawerOpen={handleDrawerOpen} />
        </div>
    )
}

export default HamburgerMenu;

