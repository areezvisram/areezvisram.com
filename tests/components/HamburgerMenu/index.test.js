import React from 'react';
import { shallow, mount } from 'enzyme';
import HamburgerMenu from '../../../src/components/HamburgerMenu';
import { Box, IconButton, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import DrawerMenu from '../../../src/components/DrawerMenu';
import { act } from 'react-dom/test-utils';

describe('<HamburgerMenu />', () => {
    let props;
    beforeEach(() => {
        props = {            
            handleDrawerOpen: jest.fn(),
            tabInfo: [{
                label: 'TEST',
                route: '/test',
                value: 1
            }],
            tabComponent: Typography
        }
    });

    it('should render as expected', () => {
        const hamburgerMenu = shallow(<HamburgerMenu {...props} />)
        expect(hamburgerMenu).toMatchSnapshot();
    });

    it('should render a Box', () => {
        const hamburgerMenu = mount(<HamburgerMenu {...props} />);
        expect(hamburgerMenu.find(Box).exists()).toBeTruthy();
    });

    describe('inside the Box', () => {
        it('should render an IconButton', () => {
            const hamburgerMenu = mount(<HamburgerMenu {...props} />);
            const iconButton = hamburgerMenu.find(Box).find(IconButton);
            expect(iconButton.exists()).toBeTruthy();
        });

        it('should render a MenuIcon inside the IconButton', () => {
            const hamburgerMenu = mount(<HamburgerMenu {...props} />);
            const menuIcon = hamburgerMenu.find(Box).find(IconButton).find(MenuIcon);
            expect(menuIcon.exists()).toBeTruthy();
        });

        it('should render a DrawerMenu component', () => {
            const hamburgerMenu = mount(<HamburgerMenu {...props} />);
            const drawerMenu = hamburgerMenu.find(Box).find(DrawerMenu);
            expect(drawerMenu.exists()).toBeTruthy();
        });

        it('should pass the appropriate props to the DrawerMenu', () => {
            const hamburgerMenu = mount(<HamburgerMenu {...props} />);
            const drawerMenu = hamburgerMenu.find(Box).find(DrawerMenu);
            expect(drawerMenu.prop('tabInfo')).toEqual(props.tabInfo);
            expect(drawerMenu.prop('tabComponent')).toEqual(props.tabComponent);
        });

        describe('state changes', () => {            
            it('should initially render the DrawerMenu closed', () => {
                const hamburgerMenu = mount(<HamburgerMenu {...props} />);
                const drawerMenu = hamburgerMenu.find(Box).find(DrawerMenu);                
                expect(drawerMenu.prop('isDrawerOpen')).toBe(false);
            });

            it('should render the DrawerMenu open when the icon is clicked', () => {
                const hamburgerMenu = mount(<HamburgerMenu {...props} />);
                const iconButton = hamburgerMenu.find(Box).find(IconButton).at(0);
                
                // Click the icon, then update the component
                act(() => {
                    iconButton.props().onClick();
                });                
                hamburgerMenu.update();

                const drawerMenu = hamburgerMenu.find(Box).find(DrawerMenu);
                expect(drawerMenu.prop('isDrawerOpen')).toEqual(true);
            });
        })
    })
})