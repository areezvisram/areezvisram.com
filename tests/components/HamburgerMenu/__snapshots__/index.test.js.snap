// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburgerMenu /> should render as expected 1`] = `ShallowWrapper {}`;
